### ISSUE TITLE
Error connecting and handling sessions in the WalletConnect SDK

### ISSUE THREAD

**Issue by tieugame2110** [WalletConnect/WalletConnectUnity#161]
After I use the mobile device to scan the QR code, it only closes the modal but does not execute the `OnAfterConnect` function

**Expected behavior**
after scan QR Code, the `WalletConnect.Instance.ActiveSessionChanged` function will be executed
note*: i unchecked `ResumeSessionOnInit` boolean

**Screenshots**
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/2897f330-701c-4645-b744-9183a1fa6dc9)

**My Code**

```
private void Start()
        {
            //Application.targetFrameRate = Screen.currentResolution.refreshRate;

            // When WalletConnectModal is ready, enable buttons and subscribe to other events.
            // WalletConnectModal.SignClient can be null if WalletConnectModal is not ready.
            WalletConnectModal.Ready += (sender, args) =>
            {
                //// SessionResumed is true if Modal resumed session from storage
                if (args.SessionResumed)
                {
                    OnAfterConnect();
                    Debug.Log(1);
                }
                else
                {
                    OnAfterDisconnect();
                    Debug.Log(2);
                }

                // Invoked after wallet connected
                WalletConnect.Instance.ActiveSessionChanged += (_, @struct) =>
                {
                    if (string.IsNullOrEmpty(@struct.Topic))
                        return;

                    Debug.Log($"[WalletConnectModalSample] Session connected. Topic: {@struct.Topic}");
                    OnAfterConnect();
                    Debug.Log(3);
                };

                // Invoked after wallet disconnected
                WalletConnect.Instance.SessionDisconnected += (_, _) =>
                {
```
                    Debug.Log($"[WalletConnectModalSample] Session deleted.");
                    OnAfterDisconnect();
                    Debug.Log(4);
                };
            };
        }

        private void OnAfterConnect()
        {
            var UWM = UIWalletManager.Instance;
            var address = GetAddress();
            UWM.web3Manager.userAddress = address;
            UWM.screenManager.ShowScreen("HOME");
            UWM.web3Manager.loginWith = LoginWith.WCModal;
            Debug.Log(address);
        }

        private void OnAfterDisconnect()
        {
            WalletConnectModal.Disconnect();
        }
**Environment**
 - OS: Android
 - Unity version 2022.3.5f1
 - Package name and version: WalletConnectUnity Modal v1.0.6

**Comment by skibitsky** [comment]
Hello @tieugame2110,

Does it log any errors? What mobile wallets did you use test it with?

**Comment by tieugame2110** [comment]
Hi @skibitsky,
Action does not provide any feedback on errors, i test with MetaMask, Spot OnChain wallet (Goerli testnet)

The notable point is that the wallet is okay the first time it connects, but after disconnecting or even deleting the session on the device, this error occurs

**Comment by skibitsky** [comment]
@tieugame2110

Can you please describe exact steps to reproduce it?

> or even deleting the session on the device

How exactly did you delete the session?

**Comment by tieugame2110** [comment]
@skibitsky 
delete the session on MetaMask:
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/79f6188b-00ab-4489-b3ad-118012286d7a)
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/5835c919-324f-4973-94d9-9bed18472d1b)
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/6be9a946-7a7b-41a3-89ec-575f9ebae076)

delete the session on Spot OnChain:
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/824abeaa-fffb-4c6e-8ab3-faf66d116417)
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/56f6f18e-9137-412d-ac82-65bdf7fc88dd)
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/ae6543ae-458f-4f9d-a900-b305756a9819)

**Comment by skibitsky** [comment]
Could you check if any of these fields in the `WalletConnectProjectConfig` scriptable asset are empty?
* `Id`
* `Name`
* `Description`
* `Url`
* `Icon` (at least one item should be present)

**Comment by tieugame2110** [comment]
@skibitsky 
I ensure that all fields have been filled in.
![image](https://github.com/WalletConnect/WalletConnectUnity/assets/43002230/1d32ba45-ff98-40c5-8012-fe976b3a3ba3)

**Comment by skibitsky** [comment]
@tieugame2110 can you access https://walletconnect.com website without VPN or any other proxy from your network?

**Comment by tieugame2110** [comment]
@skibitsky Yes, i can. I think the issue is related to the WalletConnect `session`, because any wallet when first connected executes the correct code, but from the second time onwards encounters this error.
